pkg/partmgr/partmgr.go

```diff
@@ -18,123 +18,201 @@ under the Apache 2.0 license is conditioned upon your compliance with
-	return []*DBPartition{p.headPartition}
+func (p *PartitionManager) createNewPartition(t int64) (*DBPartition, error) {
+	time := t & 0x7FFFFFFFFFFFFFF0
+	path := fmt.Sprintf("%s/%d/", p.path, time/1000)
+	partition := NewDBPartition(p, time, path)
+	p.currentPartitionInterval = partition.partitionInterval
+	p.headPartition = partition
+	p.partitions = append(p.partitions, partition)
+	err := p.updatePartitionInSchema(partition)
+	return partition, err
 }
 
-func (p *PartitionManager) GetHead() *DBPartition {
+func (p *PartitionManager) updatePartitionInSchema(partition *DBPartition) error {
+	p.cfg.Partitions = append(p.cfg.Partitions, config.Partition{StartTime: partition.startTime, SchemaInfo: p.cfg.PartitionSchemaInfo})
+	data, err := json.Marshal(p.cfg)
+	if err != nil {
+		return errors.Wrap(err, "Failed to update new partition in schema file")
+	}
+	err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path: p.path + config.SCHEMA_CONFIG, Body: data})
+	return err
+}
 
-	return p.headPartition
+func (p *PartitionManager) PartsForRange(mint, maxt int64) []*DBPartition {
+	var parts []*DBPartition
+	for _, part := range p.partitions {
+		if part.startTime >= mint && (maxt == 0 || part.startTime < maxt) {
+			parts = append(parts, part)
+		}
+	}
+	return parts
 }
 
 type DBPartition struct {
-	manager        *PartitionManager
-	path           string             // Full path (in the DB) to the partition
-	partID         int                // PartitionID
-	startTime      int64              // Start from time/date
-	days           int                // Number of days stored in the partition
-	hoursInChunk   int                // number of hours stored in each chunk
-	prefix         string             // Path prefix
-	retentionDays  int                // Keep samples for N days
-	defaultRollups aggregate.AggrType // Default Aggregation functions to apply on sample update
-	rollupTime     int64              // Time range per aggregation bucket
-	rollupBuckets  int                // Total number of buckets per partition
+	manager           *PartitionManager
+	path              string             // Full path (in the DB) to the partition
+	startTime         int64              // Start from time/date
+	partitionInterval int64              // Number of millis stored in the partition
+	chunkInterval     int64              // number of millis stored in each chunk
+	prefix            string             // Path prefix
+	retentionDays     int                // Keep samples for N hours
+	defaultRollups    aggregate.AggrType // Default Aggregation functions to apply on sample update
+	rollupTime        int64              // Time range per aggregation bucket
+	rollupBuckets     int                // Total number of buckets per partition
 }
 
 func (p *DBPartition) IsCyclic() bool {
 	return p.manager.cyclic
 }
 
-func (p *DBPartition) HoursInChunk() int {
-	return p.hoursInChunk
+// Time covered by a single chunk
+func (p *DBPartition) TimePerChunk() int64 {
```

Comment:
Isn't it `ChunkInterval`? Time is too abstract... If we decided to use same units (Milliseconds) everywhere then lets omit the time/units